src/pyinit.jl

```diff
@@ -110,6 +100,15 @@ function __init__()
                                                  t -> t.tp_call =
                                                  jl_Function_call_ptr)
 
+    # jl_FunctionType is a class, and when assigning it to an object
+    #    obj[:foo] = some_julia_function
+    # it won't behave like a regular Python method because it's not a Python
+    # function (in particular, `self` won't be passed to it). The solution is:
+    #    obj[:foo] = jlfun2pyfun(some_julia_function)
+    global const jlfun2pyfun =
+        pyeval("""lambda f: lambda *args, **kwargs: f(*args, **kwargs)""")
+    global const pyproperty = pybuiltin(:property)
```

Comment:
For global constant Python objects (any `PyObject` instance), to get proper type inference you should do `const pyproperty = PyNULL()` at the top-level scope followed by `copy!(pyproperty, pybuiltin("property"))` in `__init__`